packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -4240,13 +4246,14 @@ export class ContainerRuntime
 					contents: schemaChangeMessage,
 				};
 				this.outbox.submit({
-					contents: serializeOpContents(msg),
+					contents: this.opSerializer.stringify(msg),
```

Comment:
Maybe the ship has sailed but it seems strange to me that we stringify on this granular level (before even giving to the outbox).  I would have expected any stringification to be done just once, right before the content leaves the runtime.

Reply:
I was expecting something similar as well. Serialize right before sending the message so that it's at the very last moment. And in cases where we don't end up sending the message, no serialization / deserialization takes place.

Also, why not keep the un-serialized message in the pending state manager? That way, during resubmit, there is no need to parse it again? (We had discussed this but I can't remember).

Reply:
All of what you're both describing is where this is going.  Trying to do incremental changes.  This stuff is very entangled so even small steps will have potentially big diffs.